public/app/features/canvas/runtime/element.tsx

```diff
@@ -71,48 +65,89 @@ export class ElementState implements LayerElement {
         break;
       case VerticalConstraint.Bottom:
-        updatedPlacement.bottom = updatedPlacement.bottom ?? 0;
-        updatedPlacement.height = updatedPlacement.height ?? 100;
-        delete updatedPlacement.top;
+        placement.bottom = placement.bottom ?? 0;
+        placement.height = placement.height ?? 100;
+        style.bottom = `${placement.bottom}px`;
+        style.height = `${placement.height}px`;
+        delete placement.top;
         break;
       case VerticalConstraint.TopBottom:
-        updatedPlacement.top = updatedPlacement.top ?? 0;
-        updatedPlacement.bottom = updatedPlacement.bottom ?? 0;
-        delete updatedPlacement.height;
+        placement.top = placement.top ?? 0;
+        placement.bottom = placement.bottom ?? 0;
+        style.top = `${placement.top}px`;
+        style.bottom = `${placement.bottom}px`;
+        delete placement.height;
+        break;
+      case VerticalConstraint.Scale:
+        placement.top = placement.top ?? 0;
+        placement.bottom = placement.bottom ?? 0;
+        style.top = `${placement.top}%`;
+        style.bottom = `${placement.bottom}%`;
+        delete placement.height;
         break;
     }
 
     switch (horizontal) {
       case HorizontalConstraint.Left:
-        updatedPlacement.left = updatedPlacement.left ?? 0;
-        updatedPlacement.width = updatedPlacement.width ?? 100;
-        delete updatedPlacement.right;
+        placement.left = placement.left ?? 0;
+        placement.width = placement.width ?? 100;
+        style.left = `${placement.left}px`;
+        style.width = `${placement.width}px`;
+        delete placement.right;
         break;
       case HorizontalConstraint.Right:
-        updatedPlacement.right = updatedPlacement.right ?? 0;
-        updatedPlacement.width = updatedPlacement.width ?? 100;
-        delete updatedPlacement.left;
+        placement.right = placement.right ?? 0;
+        placement.width = placement.width ?? 100;
+        style.right = `${placement.right}px`;
+        style.width = `${placement.width}px`;
+        delete placement.left;
         break;
       case HorizontalConstraint.LeftRight:
-        updatedPlacement.left = updatedPlacement.left ?? 0;
-        updatedPlacement.right = updatedPlacement.right ?? 0;
-        delete updatedPlacement.width;
+        placement.left = placement.left ?? 0;
+        placement.right = placement.right ?? 0;
+        style.left = `${placement.left}px`;
+        style.right = `${placement.right}px`;
+        delete placement.width;
+        break;
+      case HorizontalConstraint.Scale:
+        placement.left = placement.left ?? 0;
+        placement.right = placement.right ?? 0;
+        style.left = `${placement.left}%`;
+        style.right = `${placement.right}%`;
+        delete placement.width;
         break;
     }
 
-    this.options.placement = updatedPlacement;
+    this.options.placement = placement;
+    this.sizeStyle = style;
```

Comment:
do we even need a `sizeStyle anymore?`

Reply:
It seems that to allow for custom styling of root element (width / height 100%) this structure is necessary 🤷